File: SDKs/rumble.run/2.0.3/python/src/sdk/export.py
import requests
from typing import Optional
from sdk.models import shared, operations
from . import utils

class Export:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def export_assets_csv(self, request: operations.ExportAssetsCsvRequest) -> operations.ExportAssetsCsvResponse:
        r"""Asset inventory as CSV
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/assets.csv"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportAssetsCsvResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "text/csv"):
                res.export_assets_csv_200_text_csv_binary_string = r.content
        elif r.status_code == 401:
            pass

        return res

    
    def export_assets_json(self, request: operations.ExportAssetsJSONRequest) -> operations.ExportAssetsJSONResponse:
        r"""Exports the asset inventory
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/assets.json"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportAssetsJSONResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Asset]])
                res.assets = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_assets_jsonl(self, request: operations.ExportAssetsJsonlRequest) -> operations.ExportAssetsJsonlResponse:
        r"""Asset inventory as JSON line-delimited
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/assets.jsonl"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportAssetsJsonlResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[bytes])
                res.export_assets_jsonl_200_application_json_binary_string = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_assets_nmap_xml(self, request: operations.ExportAssetsNmapXMLRequest) -> operations.ExportAssetsNmapXMLResponse:
        r"""Asset inventory as Nmap-style XML
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/assets.nmap.xml"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportAssetsNmapXMLResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "text/xml"):
                res.export_assets_nmap_xml_200_text_xml_binary_string = r.content
        elif r.status_code == 401:
            pass

        return res

    
    def export_services_csv(self, request: operations.ExportServicesCsvRequest) -> operations.ExportServicesCsvResponse:
        r"""Service inventory as CSV
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/services.csv"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportServicesCsvResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "text/csv"):
                res.export_services_csv_200_text_csv_binary_string = r.content
        elif r.status_code == 401:
            pass

        return res

    
    def export_services_json(self, request: operations.ExportServicesJSONRequest) -> operations.ExportServicesJSONResponse:
        r"""Service inventory as JSON
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/services.json"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportServicesJSONResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Service]])
                res.services = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_services_jsonl(self, request: operations.ExportServicesJsonlRequest) -> operations.ExportServicesJsonlResponse:
        r"""Service inventory as JSON line-delimited
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/services.jsonl"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportServicesJsonlResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[bytes])
                res.export_services_jsonl_200_application_json_binary_string = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_sites_csv(self, request: operations.ExportSitesCsvRequest) -> operations.ExportSitesCsvResponse:
        r"""Site list as CSV
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/sites.csv"
        
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportSitesCsvResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "text/csv"):
                res.export_sites_csv_200_text_csv_binary_string = r.content
        elif r.status_code == 401:
            pass

        return res

    
    def export_sites_json(self, request: operations.ExportSitesJSONRequest) -> operations.ExportSitesJSONResponse:
        r"""Export all sites
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/sites.json"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportSitesJSONResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Site]])
                res.sites = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_sites_jsonl(self, request: operations.ExportSitesJsonlRequest) -> operations.ExportSitesJsonlResponse:
        r"""Site list as JSON line-delimited
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/sites.jsonl"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportSitesJsonlResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[bytes])
                res.export_sites_jsonl_200_application_json_binary_string = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_wireless_csv(self, request: operations.ExportWirelessCsvRequest) -> operations.ExportWirelessCsvResponse:
        r"""Wireless inventory as CSV
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/wireless.csv"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportWirelessCsvResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "text/csv"):
                res.export_wireless_csv_200_text_csv_binary_string = r.content
        elif r.status_code == 401:
            pass

        return res

    
    def export_wireless_json(self, request: operations.ExportWirelessJSONRequest) -> operations.ExportWirelessJSONResponse:
        r"""Wireless inventory as JSON
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/wireless.json"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportWirelessJSONResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[list[shared.Wireless]])
                res.wirelesses = out
        elif r.status_code == 401:
            pass

        return res

    
    def export_wireless_jsonl(self, request: operations.ExportWirelessJsonlRequest) -> operations.ExportWirelessJsonlResponse:
        r"""Wireless inventory as JSON line-delimited
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/export/org/wireless.jsonl"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = utils.configure_security_client(self._client, request.security)
        
        r = client.request("GET", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.ExportWirelessJsonlResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[bytes])
                res.export_wireless_jsonl_200_application_json_binary_string = out
        elif r.status_code == 401:
            pass

        return res

    

File: SDKs/rumble.run/2.0.3/typescript/src/sdk/export.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse, ParamsSerializerOptions } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class Export {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * exportAssetsCsv - Asset inventory as CSV
  **/
  exportAssetsCsv(
    req: operations.ExportAssetsCsvRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportAssetsCsvResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportAssetsCsvRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/assets.csv";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportAssetsCsvResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `text/csv`)) {
                const resBody: string = JSON.stringify(httpRes?.data, null, 0);
                let out: Uint8Array = new Uint8Array(resBody.length);
                for (let i: number = 0; i < resBody.length; i++) out[i] = resBody.charCodeAt(i);
                res.exportAssetsCSV200TextCSVBinaryString = out;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportAssetsJson - Exports the asset inventory
  **/
  exportAssetsJson(
    req: operations.ExportAssetsJsonRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportAssetsJsonResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportAssetsJsonRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/assets.json";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportAssetsJsonResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.assets = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportAssetsJsonl - Asset inventory as JSON line-delimited
  **/
  exportAssetsJsonl(
    req: operations.ExportAssetsJsonlRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportAssetsJsonlResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportAssetsJsonlRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/assets.jsonl";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportAssetsJsonlResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.exportAssetsJSONL200ApplicationJSONBinaryString = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportAssetsNmapXml - Asset inventory as Nmap-style XML
  **/
  exportAssetsNmapXml(
    req: operations.ExportAssetsNmapXmlRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportAssetsNmapXmlResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportAssetsNmapXmlRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/assets.nmap.xml";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportAssetsNmapXmlResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `text/xml`)) {
                const resBody: string = JSON.stringify(httpRes?.data, null, 0);
                let out: Uint8Array = new Uint8Array(resBody.length);
                for (let i: number = 0; i < resBody.length; i++) out[i] = resBody.charCodeAt(i);
                res.exportAssetsNmapXML200TextXMLBinaryString = out;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportServicesCsv - Service inventory as CSV
  **/
  exportServicesCsv(
    req: operations.ExportServicesCsvRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportServicesCsvResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportServicesCsvRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/services.csv";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportServicesCsvResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `text/csv`)) {
                const resBody: string = JSON.stringify(httpRes?.data, null, 0);
                let out: Uint8Array = new Uint8Array(resBody.length);
                for (let i: number = 0; i < resBody.length; i++) out[i] = resBody.charCodeAt(i);
                res.exportServicesCSV200TextCSVBinaryString = out;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportServicesJson - Service inventory as JSON
  **/
  exportServicesJson(
    req: operations.ExportServicesJsonRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportServicesJsonResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportServicesJsonRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/services.json";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportServicesJsonResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.services = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportServicesJsonl - Service inventory as JSON line-delimited
  **/
  exportServicesJsonl(
    req: operations.ExportServicesJsonlRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportServicesJsonlResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportServicesJsonlRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/services.jsonl";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportServicesJsonlResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.exportServicesJSONL200ApplicationJSONBinaryString = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportSitesCsv - Site list as CSV
  **/
  exportSitesCsv(
    req: operations.ExportSitesCsvRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportSitesCsvResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportSitesCsvRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/sites.csv";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportSitesCsvResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `text/csv`)) {
                const resBody: string = JSON.stringify(httpRes?.data, null, 0);
                let out: Uint8Array = new Uint8Array(resBody.length);
                for (let i: number = 0; i < resBody.length; i++) out[i] = resBody.charCodeAt(i);
                res.exportSitesCSV200TextCSVBinaryString = out;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportSitesJson - Export all sites
  **/
  exportSitesJson(
    req: operations.ExportSitesJsonRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportSitesJsonResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportSitesJsonRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/sites.json";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportSitesJsonResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.sites = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportSitesJsonl - Site list as JSON line-delimited
  **/
  exportSitesJsonl(
    req: operations.ExportSitesJsonlRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportSitesJsonlResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportSitesJsonlRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/sites.jsonl";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportSitesJsonlResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.exportSitesJSONL200ApplicationJSONBinaryString = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportWirelessCsv - Wireless inventory as CSV
  **/
  exportWirelessCsv(
    req: operations.ExportWirelessCsvRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportWirelessCsvResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportWirelessCsvRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/wireless.csv";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportWirelessCsvResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `text/csv`)) {
                const resBody: string = JSON.stringify(httpRes?.data, null, 0);
                let out: Uint8Array = new Uint8Array(resBody.length);
                for (let i: number = 0; i < resBody.length; i++) out[i] = resBody.charCodeAt(i);
                res.exportWirelessCSV200TextCSVBinaryString = out;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportWirelessJson - Wireless inventory as JSON
  **/
  exportWirelessJson(
    req: operations.ExportWirelessJsonRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportWirelessJsonResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportWirelessJsonRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/wireless.json";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportWirelessJsonResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.wirelesses = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * exportWirelessJsonl - Wireless inventory as JSON line-delimited
  **/
  exportWirelessJsonl(
    req: operations.ExportWirelessJsonlRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ExportWirelessJsonlResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ExportWirelessJsonlRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/export/org/wireless.jsonl";
    
    const client: AxiosInstance = utils.createSecurityClient(this._defaultClient!, req.security)!;
    
    const qpSerializer: ParamsSerializerOptions = utils.getQueryParamSerializer(req.queryParams);

    const requestConfig: AxiosRequestConfig = {
      ...config,
      params: req.queryParams,
      paramsSerializer: qpSerializer,
    };
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...requestConfig,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ExportWirelessJsonlResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.exportWirelessJSONL200ApplicationJSONBinaryString = httpRes?.data;
            }
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

}
